src/utils/spawning.js

```diff
@@ -0,0 +1,68 @@
+const body = require("utils_body_constants");
+const roles = require("utils_role_constants");
+var createName = require("utils_naming")
+
+var spawner = Game.spawns['Gubbins'];
+var sources = spawner.room.find(FIND_SOURCES);
+var availableEnergy = spawner.room.energyAvailable;
+
+function spawnHarvester() {
+    sources.forEach(function(srs){
+        var tmp = spawner.room.find(FIND_MY_CREEPS, {filter: (s) => s.memory.source == srs.id})
+        if(tmp == ''){
+            targetSource = srs.id;
+        }
+        });
+
+        var bodyPieces = [WORK, MOVE, CARRY];
+        var cost = body.WORK.cost + body.MOVE.cost + body.CARRY.cost;
+        while(cost + body.WORK.cost <= availableEnergy){
+            bodyPieces.push(WORK);
+            cost += body.WORK.cost;
+        }
+
+        var code = spawner.spawnCreep(bodyPieces, createName(roles.HARVEST), {memory: {role: roles.HARVEST, source: targetSource}});
+        console.log('Spawning new harvester: ' + code);
+        return;
+}
+
```

Comment:
Please make this one empty line between functions, including of the other functions in this file.